### ISSUE TITLE
Parsing of /a{5}/u with the grammar and parsing rule in section B.1.4

### ISSUE THREAD

**Issue by nhahtdh** [tc39/ecma262#527]
According to B.1.4

> When parsing using the following grammar, each alternative is considered only if previous production alternatives do not match.

However, with the regex `/a{5}/u`, the expansion:

```
Term[U] ::
    [~U] ExtendedTerm
    [+U] Assertion[U]
    [+U] Atom[U]
    [+U] Atom[U] Quantifier
```

will try `[+U] Atom[U]` first. As a result, the regex above will be parsed to a literal sequence of `a{5}`, instead of `a` repeated 5 times, in Unicode mode.

Is this a correct understanding of the specs? Or does the clause "This alternative pattern grammar and semantics only changes the syntax and semantics of BMP patterns." mean that Unicode mode parsing should not follow the parsing rule above?

**Comment by littledan** [comment]
In Unicode mode, even with Annex B applied, `{` is not an *Atom*, as it's not in the set "*SourceCharacter* but not *SyntaxCharacter*". Therefore, the fourth production will be the only one that applies. Note that the part of the RegExp grammar that applies to Unicode RegExps is supposed to not have ambiguities (just like the main grammar) and therefore, the ordering does not matter.

**Comment by nhahtdh** [comment]
See comment on issue https://github.com/tc39/ecma262/issues/526. Anyway, it only affects 6.0 and has been fixed in ECMA 7.0.
